pkg/client/client.go

```diff
@@ -322,7 +322,7 @@ func (c *Client) Fetch(ctx context.Context, request FetchRequest) (res *FetchRes
 	c.Logger.Info("received fetch request", "disable_delete", request.DisableDataDelete, "extra_fields", request.ExtraFields)
 
 	fetchSummaries := make(chan ProviderFetchSummary, len(request.Providers))
-	errGroup, gctx := errgroup.WithContext(ctx)
+	errGroup, _ := errgroup.WithContext(ctx)
```

Comment:
Maybe add a comment regarding why we're ignoring the errgroup ctx? It's apparent now we're fixing that specific issue, but future generations will forget this and this bug might resurface. A test would've been better but it seems to be buried deep within fetch and would get complicated fast.

Reply:
Eventually this will need a refactor here, as if we aren't using gctx, I don't see much reason for the errGroup to be honest.